src/NonEmptyArray.ts

```diff
@@ -225,6 +235,16 @@ export function concat<A>(first: Array<A>, second: Array<A>): Array<A> {
   return first.concat(second)
 }
 
+/**
+ * @category combinators
+ * @since 2.11.0
+ */
+export function concatW<A, B>(first: Array<A>, second: NonEmptyArray<B>): NonEmptyArray<A | B>
+export function concatW<A, B>(first: NonEmptyArray<A>, second: Array<B>): NonEmptyArray<A | B>
+export function concatW<A, B>(first: Array<A>, second: Array<B>): Array<A | B> {
```

Comment:
IMO this should go to `Array` and curried, see also (v3): https://github.com/gcanti/fp-ts/blob/ade368f43098a141fc67ccdb7b7fc9cf728e835c/src/ReadonlyArray.ts#L239-L257

Reply:
I've had a go at backporting the changes from v3 and there seem to be type problems. Not sure what the difference is between TS 3.5 and 4.1 that seems to stop it from working. Going to back out of `concat` changes for this PR.

Reply:
@thewilkybarkid you are right, there's a type problem in v3 too, I'll look into that.

Reply:
Happy to have a go if you like.

Reply:
@thewilkybarkid fixed for now by exporting `concat` / `concatW` from [ReadonlyArray](https://github.com/gcanti/fp-ts/blob/86b8c14f78fb06a94f4337963332dfde74e563d4/src/ReadonlyArray.ts#L239-L250) and [ReadonlyNonEmptyArray](https://github.com/gcanti/fp-ts/blob/86b8c14f78fb06a94f4337963332dfde74e563d4/src/ReadonlyNonEmptyArray.ts#L204-L221)